src/languages/vbnet.js

```diff
@@ -1,63 +1,114 @@
-/*
-Language: Visual Basic .NET
-Description: Visual Basic .NET (VB.NET) is a multi-paradigm, object-oriented programming language, implemented on the .NET Framework.
-Author: Poren Chiang <ren.chiang@gmail.com>
-Website: https://docs.microsoft.com/en-us/dotnet/visual-basic/getting-started/
-*/
+/**
+ * Language: Visual Basic .NET
+ * Description: Visual Basic .NET (VB.NET) is a multi-paradigm, object-oriented programming language, implemented on the .NET Framework.
+ * Authors: Poren Chiang <ren.chiang@gmail.com>, Jan Pilzer
+ * Website: https://docs.microsoft.com/dotnet/visual-basic/getting-started/
+ */
 
 export default function(hljs) {
+  const CHAR = {
+    className: 'string',
+    begin: /"(""|[^/n])"C\b/,
+  };
+
+  const STRING = {
+    className: 'string',
+    begin: /"/,
+    end: /"/,
+    illegal: /\n/,
+    contains: [{ begin: /""/ }], // double quote escape
+  };
+
+  /**
+   * Date Literals consist of a date or a time or both separated by whitespace, surrounded by #.
+   * Date can be in M/D/YYYY or YYYY-MM-DD format.
+   * A time value may be specified either using a 24-hour value (H:mm[:ss]) or a 12-hour value (h[:mm[:ss]] A)
+   */
+  const DATE = {
+    className: 'literal',
+    begin: /# *(((\d+[-/]){2}\d+)|((\d+(:\d+){1,2} ?(AM|PM)?)|(\d+ ?(AM|PM)))|((\d+[-/]){2}\d+ ((\d+(:\d+){1,2} ?(AM|PM)?)|(\d+ ?(AM|PM))))) *#/,
+  };
+
+  /**
+   * Float: /-?\b\d[\d_]*((\.[\d_]+(E[+-]?[\d_]+)?)|(E[+-]?[\d_]+))[RFD@!#]?/
+   * Integer (base 10): /-?\b\d[\d_]*((U?[SIL])|[%&])?/
+   * Integer (base 16): /&H[\dA-F_]+((U?[SIL])|[%&])?/
+   * Integer (base  8): /&O[0-7_]+((U?[SIL])|[%&])?/
+   * Integer (base  2): /&B[01_]+((U?[SIL])|[%&])?/
+   */
+  const NUMBER = {
+    className: 'number',
+    begin: /((&[HOB])[\dA-F_]+|(-?\b\d[\d_]*))((\.[\d_]+)?(E[+-]?[\d_]+)?)?((U?[SIL])|[RFD@!#%&])?/,
```

Comment:
[Maintainability] Please beak these into separate rules using `variants` as you have them shown above.

Reply:
You can see many other grammars for examples.